app/screens/channel/channel_post_list/channel_post_list.js

```diff
@@ -61,12 +61,18 @@ export default class ChannelPostList extends PureComponent {
 
         this.setState({
             visiblePostIds,
+            loading: nextProps.channelId !== this.props.channelId,
         });
     }
 
     componentDidUpdate(prevProps) {
-        if (prevProps.channelId !== this.props.channelId && tracker.channelSwitch) {
-            this.props.actions.recordLoadTime('Switch Channel', 'channelSwitch');
+        if (prevProps.channelId !== this.props.channelId) {
+            if (tracker.channelSwitch) {
+                this.props.actions.recordLoadTime('Switch Channel', 'channelSwitch');
+            }
+            InteractionManager.runAfterInteractions(() => {
+                this.setState({loading: false});
```

Comment:
I don't think we should be setting the state in the componentDidUpdate, is there any other way to solve it? I know the condition will prevent an infinite loop, but again I'm not too comfortable with this

Reply:
didn't know it's an antipattern...moved it to componentWillReceiveProps instead